src/pure/pure.ml

```diff
@@ -158,23 +159,23 @@ let handle_command : state -> Command.t -> command_result =
         (t, ss, d.pdata_state, d.pdata_finalize, d.pdata_prv, d.pdata_sym_pos)
       in
         Cmd_Proof(ps, d.pdata_proof, d.pdata_sym_pos, d.pdata_end_pos)
-  with Fatal(Some p,m) ->
-    Cmd_Error(Some p, m)
+  with Fatal(Some p,m, _) ->
+    Cmd_Error(Some p, Pos.popt_to_string p ^ " " ^ m)
 
 let handle_tactic : proof_state -> Tactic.t -> int -> tactic_result =
   fun (_, ss, ps, finalize, prv, sym_pos) tac n ->
   try
     let ps, qres = Handle.Tactic.handle ss sym_pos prv (ps, None) tac n in
     let qres = Option.map (fun f -> f ()) qres in
     Tac_OK((Time.save (), ss, ps, finalize, prv, sym_pos), qres)
-  with Fatal(Some p,m) ->
+  with Fatal(Some p,m, _) ->
     Tac_Error(Some p, Pos.popt_to_string p ^ " " ^ m)
 
 let end_proof : proof_state -> command_result =
   fun (_, ss, ps, finalize, _, _) ->
   try Cmd_OK((Time.save (), finalize ss ps), None)
-  with Fatal(Some p,m) ->
-    Cmd_Error(Some p, m)
+  with Fatal(Some p,m, _) ->
+    Cmd_Error(Some p, Pos.popt_to_string p ^ " " ^ m)
```

Comment:
Why the position is added in the message? Isn't it going to be duplicated?
Moreover, the new third argument of Fatal is ignored: isn't it a problem?